src/default_config.ts

```diff
@@ -23,6 +23,11 @@ const DEFAULT_CONFIG = {
      */
   DEFAULT_REQUEST_TIMEOUT: 30 * 1000,
 
+    /**
+     * Default connection time after which a request will timeout, in ms.
+     * @type {Number}
+     */
+  DEFAULT_CONNECTION_TIMEOUT: 15 * 1000,
```

Comment:
How did you choose this value?

Reply:
It has to be lower than `DEFAULT_REQUEST_TIMEOUT` to make sense.
I have set it to half the `DEFAULT_REQUEST_TIMEOUT` which is 30sec, that's why the 15sec.

As a comparison the shaka-player has a default `connectionTimeout` set at 10sec.

I can change this if you think it's too high, I just wanted to not have a too agressive timeout for users that instantiate the player with the default options.